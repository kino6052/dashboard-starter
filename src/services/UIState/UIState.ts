import { IStateObject, Utils } from "../../Shared/utils";
import { createBrowserHistory } from "history";

export interface IMenuState {
    drawer: {
        open: boolean
    }
}

export interface IRouterState {
    location: AppRoute
}

export const DEFAULT_MENU_STATE: IMenuState = {
    drawer: {
        open: false
    }
}

export enum AppRoute {
    Home = '/',
    Calendar = '/calendar'
}

export const DEFAULT_ROUTER_STATE: IRouterState = {
    location: AppRoute.Home
}

export const DRAWER_WIDTH = 240;

export class UIState {
    static MenuState: IStateObject<IMenuState> = Utils.generateState(DEFAULT_MENU_STATE);
    static RouterState: IStateObject<IRouterState> = Utils.generateState(DEFAULT_ROUTER_STATE);
    static History = createBrowserHistory();
}

UIState.RouterState.subscribe(state => UIState.History.push(state.location));