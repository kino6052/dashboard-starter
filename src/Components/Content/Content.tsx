import React, { Component } from 'react';
import { Router, Route } from "react-router";
import { AppRoute, UIState, IMenuState, DRAWER_WIDTH } from '../../Services/UIState/UIState';
import { StateProps } from '../../Shared/interfaces/PropTypes';
import classNames from 'classnames';
import { createStyles, Theme, withStyles } from '@material-ui/core';
import { withState } from '../../Shared/Containers';

const styles = (theme: Theme) =>
  createStyles<string>({
    content: {
      flexGrow: 1,
      padding: theme.spacing.unit * 3,
      transition: theme.transitions.create("margin", {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.leavingScreen
      }),
      marginLeft: -DRAWER_WIDTH
    },
    contentShift: {
      transition: theme.transitions.create("margin", {
        easing: theme.transitions.easing.easeOut,
        duration: theme.transitions.duration.enteringScreen
      }),
      marginLeft: 0
    },
    drawerHeader: {
      display: "flex",
      alignItems: "center",
      padding: "0 8px",
      ...theme.mixins.toolbar,
      justifyContent: "flex-end"
    }
  });

declare type Props = StateProps<IMenuState> & ReturnType<typeof styles>;

const ContentComponent = (props: {}) => {
    const { classes, state: { drawer: { open }} } = props as Props;
    return (
        <main className={classNames(classes.content, { [classes.contentShift as string]: open })}>
            <div className={classes.drawerHeader as string} />
            <Router history={ UIState.History }>
                <Route exact path={AppRoute.Home} component={() => <div>Test 1</div>} />
                <Route exact path={AppRoute.Calendar} component={() => <div>Test 2</div>} />
            </Router>
        </main>
    )
}

export const StateContent = withState(UIState.MenuState, withStyles(styles)(ContentComponent) as unknown as new() => Component);
