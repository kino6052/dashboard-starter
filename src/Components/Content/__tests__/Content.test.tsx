import React from 'react';
import Enzyme, { mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { StateContent } from "../Content";
import { UIState, AppRoute } from "../../../Services/UIState/UIState";

Enzyme.configure({ adapter: new Adapter() });

describe('Content', () => {
    it('should correctly render content', () => {
        const wrapper = mount(<StateContent/>);
        UIState.RouterState.setState({ location: AppRoute.Home });
        expect(wrapper.find('Route').at(0)).toMatchSnapshot();
        UIState.RouterState.setState({ location: AppRoute.Calendar });
        expect(wrapper.find("Route").at(0)).toMatchSnapshot();
    })
});