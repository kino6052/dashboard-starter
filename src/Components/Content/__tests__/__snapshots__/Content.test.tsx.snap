// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content should correctly render content 1`] = `ReactWrapper {}`;

exports[`Content should correctly render content 2`] = `ReactWrapper {}`;
